pkg/agent/controller/agent.go

```diff
@@ -564,3 +579,32 @@ func (a *Controller) filterLocalEndpointSlices(obj runtime.Object, numRequeues i
 
 	return obj, false
 }
+
+func (a *Controller) getGlobalIP(service *corev1.Service) (ip, reason, msg string) {
+	if a.globalnetEnabled {
+		ip = a.getGlobalIPFromService(service)
+		if ip != "" {
+			return ip, "", ""
+		}
+
+		ingressIP := a.getIngressIP(service.Name, service.Namespace)
+		if ingressIP == nil || ingressIP.allocatedIP == "" {
+			reason, msg = getReasonAndMsgFromIngressIP(ingressIP)
+			return "", reason, msg
+		}
+
+		return ingressIP.allocatedIP, "", ""
+	}
+
+	return "", "GlobalnetDisabled", "Globalnet is not enabled"
+}
+
+func (a *Controller) getIngressIP(name, namespace string) *IngressIP {
+	obj, err := a.ingressIPClient.Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+
+	if err != nil {
+		return nil
+	}
+
+	return parseIngressIP(obj)
+}
```

Comment:
I don't we should ignore the error from `Get`. I think we should return a bool indicating found or not and an error.

```suggestion
func (a *Controller) getIngressIP(name, namespace string) (*IngressIP, bool, error) {
    obj, err := a.ingressIPClient.Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
    if apierrors.IsNotFound(err) {
        return nil, false, nil 
    }
        
    if err != nil {
        return nil, false, err
    }
     
    return parseIngressIP(obj), true, nil
}
```